File: 1518.water-bottles.cpp
/*
 * @lc app=leetcode id=1518 lang=cpp
 *
 * [1518] Water Bottles
 */

// @lc code=start
class Solution
{
public:
    int numWaterBottles(int numBottles, int numExchange)
    {
        // return numBottles + (numBottles - 1) / (numExchange - 1);
        int res = numBottles;
        int rest_bottles = numBottles;
        while (rest_bottles >= numExchange)
        {
            int exchanged_bottles = rest_bottles / numExchange;
            res += exchanged_bottles;
            rest_bottles = rest_bottles % numExchange + exchanged_bottles;
        }
        return res;
    }
};
// @lc code=end

File: py/1518.water-bottles.py
#
# @lc app=leetcode id=1518 lang=python3
#
# [1518] Water Bottles
#

# @lc code=start
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        # return numBottles + (numBottles - 1) // (numExchange - 1);
        res = numBottles
        rest_bottles = numBottles
        while rest_bottles >= numExchange:
            exchanged_bottles = rest_bottles // numExchange
            res += exchanged_bottles
            rest_bottles = rest_bottles % numExchange + exchanged_bottles
        return res
# @lc code=end
